pkgs/development/python-modules/sphinxcontrib-moderncmakedomain/default.nix

```
@@ -0,0 +1,41 @@
+{
+  lib,
+  buildPythonPackage,
+  defusedxml,
+  fetchPypi,
+  hatchling,
+  pytestCheckHook,
+  sphinx,
+}:
+
+buildPythonPackage rec {
+  pname = "sphinxcontrib-moderncmakedomain";
+  version = "3.29.0";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit version;
+    pname = "sphinxcontrib_moderncmakedomain";
+    hash = "sha256-NYfe8kH/JXfQu+8RgQoILp3sG3ij1LSgZiQLXz3BtbI=";
+  };
+
+  nativeBuildInputs = [
+    hatchling
+    sphinx
+  ];
```

```suggestion
  build-system = [ hatchling ];
  
  dependencies = [ sphinx ];
```
We use `build-system` and `dependencies` for python dependencies.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

Also, following the upstream, `sphinx` should go to `dependencies`.
https://github.com/scikit-build/moderncmakedomain/blob/b4cdb0089ad40189f4cffeab7f859265da599336/pyproject.toml#L40-L42